File: apps/web/src/api/catalog.ts
import { request } from './client'
import type { CatalogExercise, MuscleGroup } from '@/types/api'

export interface CatalogListParams {
  search?: string
  muscle_group?: MuscleGroup
}

// catalogApi reads the shared, read-only exercise catalog (PRD 0006). The list
// endpoint wraps results under `exercises`, matching the routines/sessions
// list convention.
export const catalogApi = {
  list(params: CatalogListParams = {}) {
    const qs = new URLSearchParams()
    if (params.search) qs.set('search', params.search)
    if (params.muscle_group) qs.set('muscle_group', params.muscle_group)
    const query = qs.toString()
    const path = query ? `/api/v1/exercise-catalog?${query}` : '/api/v1/exercise-catalog'
    return request<{ exercises: CatalogExercise[] }>(path).then((r) => r.exercises)
  },
}

File: apps/api/internal/handler/catalog.go
package handler

import (
	"context"
	"net/http"

	"github.com/irfanmaulana007/gym-assistant/apps/api/internal/domain"
	"github.com/irfanmaulana007/gym-assistant/apps/api/pkg/httpx"
)

type catalogService interface {
	List(ctx context.Context, search, muscleGroup string) ([]domain.CatalogExercise, error)
}

// CatalogHandler serves the read-only shared exercise catalog (PRD 0006).
type CatalogHandler struct {
	svc catalogService
}

// NewCatalogHandler builds a CatalogHandler.
func NewCatalogHandler(svc catalogService) *CatalogHandler {
	return &CatalogHandler{svc: svc}
}

// List returns catalog entries, honoring optional ?search= and ?muscle_group=
// query filters.
func (h *CatalogHandler) List(w http.ResponseWriter, r *http.Request) {
	search := r.URL.Query().Get("search")
	muscleGroup := r.URL.Query().Get("muscle_group")
	entries, err := h.svc.List(r.Context(), search, muscleGroup)
	if err != nil {
		httpx.WriteError(w, err)
		return
	}
	httpx.JSON(w, http.StatusOK, map[string]any{"exercises": entries})
}
